src/main/java/de/dennisguse/opentracks/fragments/MarkerDetailFragment.java

```diff
@@ -170,8 +173,14 @@ public void setMenuVisibility(boolean menuVisible) {
     @Override
     public void onCreateOptionsMenu(@NonNull Menu menu, MenuInflater inflater) {
         inflater.inflate(R.menu.marker_detail, menu);
-
+        shareMarkerImageMenuItem = menu.findItem(R.id.marker_detail_share);
         updateWaypoint(false);
+        updateMenuItems();
+    }
+
+    private void updateMenuItems() {
+        if (shareMarkerImageMenuItem != null)
```

Comment:
Is this null check required?
And this code could actually be moved into the calling methods (as long as it is only one menuItem).

Reply:
Yes, it's required because I need to check if this option must be hidden every time this MarkerDetailFragment is resumed.

The first time the user enter in the marker detail, MarkerDetailFragment onResume is execute and the menu isn't created yet so shareMarkerImageMenuItem is null.

But now you imagine that you go to the detail of a marker with image. Then you edit the marker and deletes the image. Later you go back to the MarkerDetailFragment, onResume is executed, shareMarkerImageMenuItem isn't null and it must be not visible because the image was deleted.